hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/TaskletExecutionService.java

```diff
@@ -270,6 +262,7 @@ public void run() {
                 tracker.executionTracker.exception(new JetException("Exception in " + t + ": " + e, e));
             } finally {
                 blockingWorkerCount.decrementAndGet();
+                UserMetricsImpl.getContainer().setContext(null);
```

Comment:
the logic here is racy. if the setContext() throws then count will get decretmented without having been incremented.